workspaces/ui-v2/src/optic-components/pages/diffs/ReviewEndpointDiffPage/ReviewEndpointDiffPage.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { FC, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Helmet } from 'react-helmet';
+import { TwoColumnFullWidth } from '../../../layouts/TwoColumnFullWidth';
+import { DiffHeader } from '../../../diffs/DiffHeader';
+import { DiffCard } from '../../../diffs/render/DiffCard';
+import { makeStyles } from '@material-ui/styles';
+import { AddContribution } from '../../../../lib/command-factory';
+
+import { EndpointDocumentationPane } from '../EndpointDocumentationPane';
+import { useSharedDiffContext } from '../../../hooks/diffs/SharedDiffContext';
+import { Collapse, IconButton, Typography } from '@material-ui/core';
+import { ArrowLeft, ArrowRight } from '@material-ui/icons';
+import MenuIcon from '@material-ui/icons/Menu';
+import { SimulatedCommandStore } from '../../../diffs/contexts/SimulatedCommandContext';
+import {
+  EndpointName,
+  EditableTextField,
+  TextFieldVariant,
+} from '../../../common';
+import { IEndpoint } from '../../../hooks/useEndpointsHook';
+import { IForkableSpectacle } from '@useoptic/spectacle';
+import { useDiffReviewCapturePageLink } from '../../../navigation/Routes';
+import { getEndpointId } from '../../../utilities/endpoint-utilities';
+import { DiffLinks } from './DiffLinks';
+import { IInterpretation } from '../../../../lib/Interfaces';
+import { useDebouncedFn, useStateWithSideEffect } from '../../../hooks/util';
+
+const useRedirectForDiffCompleted = (shapeDiffs: IInterpretation[]) => {
+  const history = useHistory();
+  const diffReviewPage = useDiffReviewCapturePageLink();
+  const { isDiffHandled } = useSharedDiffContext();
+
+  useEffect(() => {
+    if (
+      shapeDiffs.length > 0 &&
+      shapeDiffs.every((i) => isDiffHandled(i.diffDescription?.diffHash!))
+    ) {
+      history.push(diffReviewPage.linkTo());
+    }
+  }, [shapeDiffs, diffReviewPage, history, isDiffHandled]);
+};
+
+type ReviewEndpointDiffPageProps = {
+  endpoint: IEndpoint;
+  spectacle: IForkableSpectacle;
+  shapeDiffs: IInterpretation[];
+  method: string;
+  pathId: string;
+};
+
+export const ReviewEndpointDiffPage: FC<ReviewEndpointDiffPageProps> = ({
```

Comment:
I _usually_ prefer to have 1 component per file (but I try not to be dogmatic about it) - I didn't do this here since I'm not sure what eveyone's thoughts are around having component "folders" for each page (or subpages)